### PR TITLE
perf(TextBox): Avoid unnecessary allocations

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.RuntimeTests/Extensions/TransformExtensions.Android.cs [hunk 1]
```diff
@@ -6,22 +6,10 @@
...
-		internal static DependencyPropertyChangedEventArgs CreateInitialChangedEventArgs(this Transform transform, DependencyProperty property)
-		{
-			return new DependencyPropertyChangedEventArgs(
-				property,
-				null,
-				DependencyPropertyValuePrecedences.DefaultValue,
-				transform.GetValue(property),
-				transform.GetCurrentHighestValuePrecedence(property)
-			);
-		}
```
Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/Extensions/TransformExtensions.Android.cs:22]: I was looking for `DependencyPropertyChangedEventArgs` constructor usages and found this. This method was already unused so I deleted it.

The remaining method in this class is only used by runtime tests so I moved the class.

**Review by Youssef1313 — COMMENTED** [review]
@jeromelaban Are we okay with the minor breaking changes here making into master or should I move this to breaking changes branch?

**Review by jeromelaban — COMMENTED** [review]
This change will likely not save much, as the original propertychanged args are generated anyways. It will only save property accessors, which are generally inlined.

**Comment by Youssef1313** [comment]
> as the original propertychanged args are generated anyways

I'm not sure I'm following. But to clarify, the allocation I'm referring to happens in https://github.com/unoplatform/uno/blob/fe75d764ca18a10ef1ccb465efbf7f30015bb209/src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs#L189 and is now completely removed.

We could be allocating *similar* objects in DependencyObjectStore, but they are different allocations.

Though I don't have benchmarks showing whether this is beneficial, but the change is quite straightforward.

We can close if this is unlikely to be an improvement.

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs [hunk 2]
```diff
@@ -376,29 +374,25 @@ private void UpdateInputScope(InputScope inputScope)
...
-			var acceptsReturn = (bool)e.NewValue;
-			_textBoxView?.SetHorizontallyScrolling(!acceptsReturn);
+			_textBoxView?.SetHorizontallyScrolling(!newValue);
 			_textBoxView?.UpdateSingleLineMode();
 
 			UpdateInputScope(InputScope);
 		}
 
-		partial void OnTextWrappingChangedPartial(DependencyPropertyChangedEventArgs e)
+		partial void OnTextWrappingChangedPartial(TextWrapping newValue)
```
Comment (spouliot) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs:393]: the `newValue` argument is not used and could be removed

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs [hunk 3]
```diff
@@ -421,7 +415,7 @@ private void UpdateInputScope(InputScope inputScope)
 			}
 		}
 
-		partial void OnIsReadonlyChangedPartial(DependencyPropertyChangedEventArgs e) => UpdateTextBoxViewReadOnly();
+		partial void OnIsReadonlyChangedPartial(bool newValue) => UpdateTextBoxViewReadOnly();
```
Comment (spouliot) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs:418]: `newValue` is unused

**Review by spouliot — COMMENTED** [review]
There are other cases (beside the two I identified) where the arguments are unused. It would likely be beneficial to remove them.

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs (hunk 2 above)
Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs:393]: Sure, can remove them. But let's first see if we even want to take this PR or not.

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs [hunk 4]
```diff
@@ -531,12 +529,12 @@ public InputScope InputScope
...
 						}
 					},
-					propertyChangedCallback: (s, e) => ((TextBox)s)?.OnInputScopeChanged(e)
+					propertyChangedCallback: (s, e) => ((TextBox)s)?.OnInputScopeChanged((InputScope)e.NewValue)
 				)
 			);
 
-		protected void OnInputScopeChanged(DependencyPropertyChangedEventArgs e) => OnInputScopeChangedPartial(e);
-		partial void OnInputScopeChangedPartial(DependencyPropertyChangedEventArgs e);
+		protected void OnInputScopeChanged(InputScope newValue) => OnInputScopeChangedPartial(newValue);
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs:536]: I think these should be `private` not `protected`.

**Comment by jeromelaban** [comment]
> I'm not sure I'm following. But to clarify, the allocation I'm referring to happens in

I'm not seeing any use of `CreateInitialChangedEventArgs`, maybe my search is failing me?

Most changes done in this PR are about changing the parameters of methods invoked from PropertyChanged handlers which already have created the event args instances. This means that aside from reducing property invocations in the handlers themselves, I'm not seeing the gain in performance.

If performance does not change, I'd rather keep original source compatibility.

**Comment by Youssef1313** [comment]
> I'm not seeing any use of `CreateInitialChangedEventArgs`, maybe my search is failing me?

The allocations being removed are in `CreateInitialValueChangerEventArgs` not `CreateInitialChangedEventArgs`

They happen in:

https://github.com/unoplatform/uno/blob/6627eb8a2264f0ca6e4aa922941b09afabe34c8d/src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs#L128-L131
https://github.com/unoplatform/uno/blob/6627eb8a2264f0ca6e4aa922941b09afabe34c8d/src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs#L135
https://github.com/unoplatform/uno/blob/6627eb8a2264f0ca6e4aa922941b09afabe34c8d/src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs#L137-L139
https://github.com/unoplatform/uno/blob/6627eb8a2264f0ca6e4aa922941b09afabe34c8d/src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs#L142

and

https://github.com/unoplatform/uno/blob/6627eb8a2264f0ca6e4aa922941b09afabe34c8d/src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.macOS.cs#L25

https://github.com/unoplatform/uno/blob/6627eb8a2264f0ca6e4aa922941b09afabe34c8d/src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.iOS.cs#L27

**Comment by Youssef1313** [comment]
> > The allocations being removed are in `CreateInitialValueChangerEventArgs` not `CreateInitialChangedEventArgs`
> 
> I see it now, thanks for the update. So we're mostly avoiding allocations on the creation of the control, not during the rest lifecycle of the control. It's good to take, and this code is not originating from WinUI's source, so we can keep it the way you changed it.

Yes, but little more than on creation because `UpdateTextBoxView` calls `InitializeProperties` on some platforms, and `UpdateTextBoxView` is called on some property changes. So, we avoid allocations for control creation plus very few property changes on specific platforms.

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**